server/channels/web/oauth.go

```diff
@@ -358,6 +395,15 @@ func loginWithOAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	loginHint := r.URL.Query().Get("login_hint")
 	redirectURL := r.URL.Query().Get("redirect_to")
+	desktopToken := r.URL.Query().Get("desktop_token")
+	if desktopToken != "" {
+		desktopTokenErr := c.App.SaveClientDesktopToken(desktopToken, time.Now().Unix())
```

Comment:
We don't need to save the desktop token on the server side.  The desktop token will be generated in the desktop, passed as a query parameter to the external idP and then returned to the desktop app via deeplink once the user has been redirected to the landing page.  It's the desktop that needs to store the desktop token so that it can verify it matches the original copy once it receives it again.